pkg/capi/gcp.go

```diff
@@ -0,0 +1,295 @@
+	}
+
+	for diskTypeName := range diskTypes {
+		// OCP-77825 - instances support disks pd-standard and pd-ssd
+		// author: miyadav@redhat.com
+		It(fmt.Sprintf("should be able to run a machine with disk type: %s", diskTypeName), func() {
+			mapiProviderSpec := getGCPMAPIProviderSpec(cl)
+			Expect(mapiProviderSpec).ToNot(BeNil())
+			mapiProviderSpec.OnHostMaintenance = "Migrate"
+			mapiProviderSpec.ConfidentialCompute = "Disabled"
+			mapiProviderSpec.Disks[0].Type = diskTypeName
+			gcpMachineTemplate = createGCPMachineTemplate(cl, mapiProviderSpec)
+			machineSet, _ = framework.CreateCAPIMachineSet(ctx, cl, framework.NewCAPIMachineSetParams(
+				"gcp-machineset",
+				clusterName,
+				mapiMachineSpec.Zone,
+				1,
+				corev1.ObjectReference{
+					Kind:       "GCPMachineTemplate",
+					APIVersion: infraAPIVersion,
+					Name:       gcpMachineTemplateName,
+				},
+			))
+			Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI machineset")
+			framework.WaitForCAPIMachinesRunning(framework.GetContext(), cl, machineSet.Name)
+		})
+	}
+
+	// OCP-74795 - add support for shielded VMs - It takes defaults if configs are not supported; eg-vtpm alone set to Enabled will result in IntergrityMonitoring also as Enabled
+	// doesn't matter what we pass - all enabled/secureboot only disabled(default)/secureboot only enabled are valid changes which we can apply
+	// author: miyadav@redhat.com
+	DescribeTable("should configure Shielded VM options correctly",
+		func(enableSecureBoot mapiv1.SecureBootPolicy, enableVtpm mapiv1.VirtualizedTrustedPlatformModulePolicy, enableIntegrityMonitoring mapiv1.IntegrityMonitoringPolicy) {
+			mapiProviderSpec := getGCPMAPIProviderSpec(cl)
+			Expect(mapiProviderSpec).ToNot(BeNil())
+			mapiProviderSpec.OnHostMaintenance = "Migrate"
+			mapiProviderSpec.ConfidentialCompute = "Disabled"
+			// Setting Shielded VM configuration
+			mapiProviderSpec.ShieldedInstanceConfig = mapiv1.GCPShieldedInstanceConfig{
+				SecureBoot:                       enableSecureBoot,
+				VirtualizedTrustedPlatformModule: enableVtpm,
+				IntegrityMonitoring:              enableIntegrityMonitoring,
+			}
+			gcpMachineTemplate = createGCPMachineTemplate(cl, mapiProviderSpec)
+			machineSet, err = framework.CreateCAPIMachineSet(ctx, cl, framework.NewCAPIMachineSetParams(
+				"gcp-machineset-shieldedvm",
+				clusterName,
+				mapiMachineSpec.Zone,
+				1,
+				corev1.ObjectReference{
+					Kind:       "GCPMachineTemplate",
+					APIVersion: infraAPIVersion,
+					Name:       gcpMachineTemplateName,
+				},
+			))
+			Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI machineset with Shielded VM config")
+
+			framework.WaitForCAPIMachinesRunning(framework.GetContext(), cl, machineSet.Name)
+
+			By("Verifying the Shielded VM configuration on the created GCP MachineTemplate")
+			createdTemplate := &gcpv1.GCPMachineTemplate{}
+			Expect(cl.Get(ctx, client.ObjectKey{
```

Comment:
See Dam's comment:
Lets' use Evenutally on all cl. (client) actions instead of Expect, so we allow for retry on brief API/network hiccups.
https://github.com/openshift/cluster-api-actuator-pkg/pull/337#discussion_r1839904474